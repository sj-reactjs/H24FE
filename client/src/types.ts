export type Category = {
    name: string
    categoryArticles: CategoryArticle
    articleCount: number
    childrenCategories: ChildCategory[]
}

export type Article = {
    name: string
    variantName: string
    prices: Prices
    images: Image[]
}

export type ChildCategory = {
    name: string
    urlPath: string
}

export type Prices = {
    currency: string
    regular: {
        value: number
    }
}

export type Image = {
    path: string
}

export type CategoryArticle = {
    articles: Article[]
}

export type PageProps = {
    categoryArticles: CategoryArticle
    articleCount: number
    childrenCategories: ChildCategory[]
    locale: string;
    name: string;
}

export type SidebarProps = {
    categories: ChildCategory[];
};