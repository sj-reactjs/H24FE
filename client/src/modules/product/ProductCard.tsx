import React from 'react';
import Flex from '../../components/Flex';
import Box from '../../components/Box';
import Text from '../../components/Text';
import Button from '../../components/Button';
import { formatter } from '../../utils/formatter';
import { Article } from '../../types'

const ProductCard: React.FC<Article> = ({
  images,
  name,
  prices
}) => {
  const handleAddtoCart = (event: any) => {
    console.log("selected Product to `add to cart`", name)
  }
  return (
    <Box
      borderWidth={'sm'}
      borderStyle={'solid'}
      borderColor={'primary_500'}
      m={'-2px'}
      borderRadius="md"
      boxShadow="lg">
      <Box role="img"
        aria-label={name}
        borderRadius="2px 2px 0px 0px"
        overflow="hidden"
        height="auto">
        <img src={images[0].path} alt={name} width="100%" />
      </Box>

      <Box p="xs" height="75px">
        <Text data-testid="product-name" fontWeight="bold" fontFamily="primary" fontSize="md">
          {name}
        </Text>
      </Box>
      <Flex p="xs">
        <Box flex="50%" alignSelf="center">
          <Text data-testid="product-price" color="primary_600">{formatter.format(prices.regular.value / 100)}</Text>
        </Box>
        <Box flex="50%">
          <Button
            tabIndex="0"
            role="button"
            aria-label={`Add ${name} to cart`}
            aria-pressed={true}
            data-testid="add-button"
            onClick={handleAddtoCart}
            variant="secondary"
            width="100%"
            p="0"
            disabled={false}>
            Add to cart
          </Button>
        </Box>
      </Flex>
    </Box>
  );
}

export default ProductCard;
