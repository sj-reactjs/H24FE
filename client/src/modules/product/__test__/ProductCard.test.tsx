import * as React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import ProductCard from '../ProductCard'
import { Article } from '../../../types';

describe('ProductCard Component', () => {
  let props: Article;
  beforeEach(() => {
    props = {
      "name": "Premium Komfortmatratze Smood",
      "variantName": "180 x 200cm",
      "images": [{
        "path": "https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp"
      }],
      "prices": {
        "currency": "EUR", "regular": {
          "value": 56999
        }
      }
    }
  })

  test('render', () => {
    const { asFragment } = render(<ProductCard {...props} />);
    expect(asFragment()).toMatchSnapshot();
  })

  test('click on add button', () => {
    render(<ProductCard {...props} />);
    expect(screen.getByTestId('add-button')).toBeInTheDocument()
    expect(screen.getByTestId('product-price')).toBeInTheDocument()
    expect(screen.getByTestId('product-name')).toBeInTheDocument()

    fireEvent.click(screen.getByTestId('add-button'))

  })

})
