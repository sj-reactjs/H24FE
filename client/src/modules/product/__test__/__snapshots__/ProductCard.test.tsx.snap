// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL bdBpPQ"
  >
    <div
      aria-label="Premium Komfortmatratze Smood"
      class="sc-bdvvtL dWIBzz"
      height="auto"
      overflow="hidden"
      role="img"
    >
      <img
        alt="Premium Komfortmatratze Smood"
        src="https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp"
        width="100%"
      />
    </div>
    <div
      class="sc-bdvvtL fBFXZO"
      height="75px"
    >
      <div
        class="sc-dkPtRN doDUdL"
        data-testid="product-name"
        font-family="primary"
        font-size="md"
        font-weight="bold"
      >
        Premium Komfortmatratze Smood
      </div>
    </div>
    <div
      class="sc-bdvvtL sc-gsDKAQ bLfgpG"
      display="flex"
    >
      <div
        class="sc-bdvvtL eTFxBV"
      >
        <div
          class="sc-dkPtRN jNYixi"
          color="primary_600"
          data-testid="product-price"
        >
          569,99 €
        </div>
      </div>
      <div
        class="sc-bdvvtL bVIZhX"
      >
        <button
          aria-label="Add Premium Komfortmatratze Smood to cart"
          aria-pressed="true"
          class="sc-hKwDye foDqhF"
          data-testid="add-button"
          role="button"
          tabindex="0"
          width="100%"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
